src/content/BatchRequestContentCollection.ts

```diff
@@ -0,0 +1,140 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @module BatchRequestContentCollection
+ */
+
+import { ErrorMappings, RequestAdapter, RequestInformation } from "@microsoft/kiota-abstractions";
+import { BatchItem, convertRequestInformationToBatchItem } from "./BatchItem";
+import { BatchResponseContent } from "./BatchResponseContent";
+import { BatchRequestContent } from "./BatchRequestContent";
+
+/**
+ * The default limit for the number of requests in a single batch.
+ */
+const DEFAULT_BATCH_LIMIT = 4;
+
+/**
+ * Represents the content of a batch request.
+ */
+export class BatchRequestContentCollection {
+  /**
+   * @private
+   * @static
+   * Executes the requests in the batch request content
+   */
+  private readonly requestAdapter: RequestAdapter;
+
+  /**
+   * @private
+   * @static
+   * Error mappings to be used while deserializing the response
+   */
+  private readonly errorMappings: ErrorMappings;
+
+  /**
+   * @private
+   * @static
+   * The maximum number of requests that can be sent in a single batch request
+   */
+  private readonly batchLimit: number;
+
+  /**
+   * @private
+   * @static
+   * The list of requests to be sent in the batch request
+   */
+  public readonly batchRequests: BatchItem[] = [];
```

Comment:
I believe this should be a collection of `BatchRequestContent`. As batch items are added, you fill them up into `BatchRequestContent` so that the user has no limit on the number of requests they make.
https://github.com/microsoftgraph/msgraph-sdk-design/blob/main/content/BatchRequestContentCollection.md#objectives

Reply:
Isn't it cleaner to hold a single list for the request. I believe operations such as add / remove elements to the array are much faster. Having a `BatchRequestContent` introduces a larger object in memory and I can't see a clear benefit of holding it here rather than chunking just before posting